app/assets/javascripts/components/common/wiki_select.jsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import createReactClass from 'create-react-class';
+import AsyncSelect from 'react-select/lib/Async';
+import PropTypes from 'prop-types';
+
+const options = [];
+/**
+ *  A Wiki Selector Component that combines both language and project into a singular searchable
+ *  component that works for both single-wiki and multi-wiki.
+ */
+const WikiSelect = createReactClass({
+  propTypes: {
+    /**
+     *  If true multiple wiki can be selected.
+     */
+    multi: PropTypes.bool,
+    /**
+     *  callback(wiki); where wiki is { language, project } if multi = false else Array of { language, project }
+     */
+    onChange: PropTypes.func,
+    /**
+     *  Custom styles for the Select Widget.
+     */
+    styles: PropTypes.object,
+    /**
+     *  An array of { language, project }
+     */
+    wikis: PropTypes.array
+  },
+
+  render() {
+    if (options.length === 0) {
+      // cache the options so it doesn't run on every render
+      const languages = JSON.parse(WikiLanguages);
+      const projects = JSON.parse(WikiProjects).filter(proj => proj !== 'wikidata');
+      for (let i = 0; i < languages.length; i += 1) {
+        for (let j = 0; j < projects.length; j += 1) {
+          const language = languages[i];
+          const project = projects[j];
+          options.push({ value: { language, project }, label: `${language}.${project}.org` });
+        }
+      }
```

Comment:
A couple things about this!

1. Sage and I were discussing and we think actually _all_ of this could separated out to a different file somewhere else. None of this code depends upon the context of the WikiSelect component, so it'd be worthwhile to just move it elsewhere and import it into this file.

1. Typically convention is to use native array methods and more "explanatory" methods and fall back to using `for` loops when you need to for a specific reason (e.g. performance). Unless you think the performance here could get particularly bad, it may be worthwhile to refactor to something like:
```js
const options = languages.map(language => {
  return projects.map(project => {
    const value = { language, project };
    const label = `${language}.${project}.org`;
    return { value, label }
  })
}).reduce((a, b) => a.concat(b)) // Flatten
```

Reply:
Actually, the WikiLanguages and WikiProjects global variable that gets defined in a HAML file is the reason I put them inside this component as refactoring this code outside of the component made a smoketest fail.